common/lib/xmodule/xmodule/x_module.py

```diff
@@ -1494,6 +1494,29 @@ def publish(self, block, event_type, event):
         # A stub publish method that doesn't emit any events from XModuleDescriptors.
         pass
 
+    def service(self, block, service_name):
+        """
+        Runtime-specific override for the XBlock service manager.  If a service is not currently
+        instantiated and is declared as a critical requirement, an attempt is made to load the
+        module.
+
+        Arguments:
+            block (an XBlock): this block's class will be examined for service
+                decorators.
+            service_name (string): the name of the service requested.
+
+        Returns:
+            An object implementing the requested service, or None.
+        """
+        # getting the service from parent module. making sure of block service declarations.
+        service = super(DescriptorSystem, self).service(block=block, service_name=service_name)
+        # Passing the block to service if it is callable e.g. ModuleI18nService. It is the responsibility of calling
+        # service to handle the passing argument.
+        # Here we are assuming that, callable service would be a ModuleI18nService and requesting "i18n" service.
+        if callable(service) and service_name == 'i18n':
```

Comment:
@cpennington  @mattdrayer take a look please. 

Tests failure issue was related to this check. It was executing in case of other services rather then `Modulei18nService`. I seems me that they are mocking other services during tests which are callable. 
I just added a `service_name` check here.